### PR TITLE
TRUNK-4705 / HTML-594 Module must be fixed to work with OpenMRS 1.12

### PR DESCRIPTION
**Pull request by rkorytkowski**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/htmlformentry/db/hibernate/HibernateHtmlFormEntryDAO.java [hunk 1]
```diff
@@ -25,9 +26,9 @@
 
 	private static Log log = LogFactory.getLog(HibernateHtmlFormEntryDAO.class);
 	
-    private SessionFactory sessionFactory;
+    private DbSessionFactory sessionFactory;
```
Comment (rkorytkowski) [inline api/src/main/java/org/openmrs/module/htmlformentry/db/hibernate/HibernateHtmlFormEntryDAO.java]: SessionFactory should be replaced with DbSessionFactory for any DAO.

**Comment by dkayiwa** [comment]
@rkorytkowski do you remember what we decided on this? :)

**Comment by mogoodrich** [comment]
I should remember, but I don't 100%.  We had a design call discussion about supporting Hibernate 4.0, and made some decisions there--I would hope the decisions are in the design call notes?

**Comment by rkorytkowski** [comment]
@dkayiwa implemented it using an alternative solution at https://github.com/openmrs/openmrs-module-xforms/commit/5d79f327d27777a40ec755a90433e4dc69067e4a

**Comment by mogoodrich** [comment]
So assumedly we have to do something similar to what was done in XForms for HFE? Can we add a note to that effect on the ticket?  

https://issues.openmrs.org/browse/HTML-594

@rkorytkowski @dkayiwa 

Thanks guys!
Mark

**Comment by mogoodrich** [comment]
So @dkayiwa, it looks like this should be ready to go?

**Comment by dkayiwa** [comment]
@mogoodrich as long as you have no objections :)

**Comment by mogoodrich** [comment]
Well, I just spent a couple hours fixing a bug caused by making this change to the reporting module, so a good time to tackle this one as well... :)  I will try it out shortly, and close this out if all goes well...

**Comment by dkayiwa** [comment]
Are these the reporting module changes that you are referring to? https://github.com/openmrs/openmrs-module-reporting/pulls

**Comment by mogoodrich** [comment]
@dkayiwa yes... the reportingrest module needed to be updated to reflect the changes as well

**Comment by dkayiwa** [comment]
I thought they are not yet merged! Are they? :)

On Thu, Nov 19, 2015 at 10:38 PM, Mark Goodrich <notifications@github.com>
wrote:

> @dkayiwa <https://github.com/dkayiwa> yes... the reportingrest module
> needed to be updated to reflect the changes as well
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-module-htmlformentry/pull/86#issuecomment-158171773>
> .
>



-- 
If we keep uppermost in our minds the unkind and unjust acts of others, we
shall find it impossible to love them as Christ has loved us; but if our
thoughts dwell upon the wondrous love and pity of Christ for us, the same
spirit will flow out to others.

**Comment by mogoodrich** [comment]
Mike merged the reporting pull request this morning.

**Comment by dkayiwa** [comment]
Oh i see!

On Thu, Nov 19, 2015 at 10:49 PM, Mark Goodrich <notifications@github.com>
wrote:

> Mike merged the reporting pull request this morning.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-module-htmlformentry/pull/86#issuecomment-158174516>
> .
>



-- 
If we keep uppermost in our minds the unkind and unjust acts of others, we
shall find it impossible to love them as Christ has loved us; but if our
thoughts dwell upon the wondrous love and pity of Christ for us, the same
spirit will flow out to others.

**Pull request merged**